File: project/backend/src/models/dtos.py
from pydantic import BaseModel, Field

class CommentDto(BaseModel):
    file_path: str = Field(alias="filePath")
    line_number: int = Field(alias="lineNumber")
    text: str = Field()

    class Config:
        validate_by_name = True

class RepositoryDto(BaseModel):
    identifier: int = Field(..., alias="identifier")
    type: str = Field(..., alias="type")
    repo_landing_page_url: str = Field(..., alias="repoLandingPageUrl")

class ProjectDto(BaseModel):
    identifier: int = Field(..., alias="identifier")
    version: str = Field(..., description="Version of the project")
    label: str = Field(..., description="Label for the project")
    read_api_url: str = Field(..., alias="readApiUrl")
    write_api_url: str = Field(None, alias="writeApiUrl", description="Optional write API URL for the project")
    repository: RepositoryDto = Field(..., description="Repository details associated with the project")

File: project/backend/src/models/dtos.ts
import { z } from "zod";

export const CommentDtoSchema = z.object({
	filePath: z.string().min(1, "File path is required"),
	lineNumber: z.number().int().positive("Line number must be a positive integer"),
	content: z.string().min(1, "Comment content is required"),
	tags: z.array(z.string()).optional(),
});

export const RepositoryDtoSchema = z.object({
	identifier: z.number().int(),
	type: z.string(),
	repoLandingPageUrl: z.string().url(),
});

export const ProjectDtoSchema = z.object({
	identifier: z.number().int(),
	version: z.string(),
	label: z.string(),
	readApiUrl: z.string().url(),
	writeApiUrl: z.string().url().nullable(),
	repository: RepositoryDtoSchema,
});

export type CommentDto = z.infer<typeof CommentDtoSchema>;
export type ProjectDto = z.infer<typeof ProjectDtoSchema>;
export type RepositoryDto = z.infer<typeof RepositoryDtoSchema>;
